appcelerator/titanium_mobile
# fix(android): Modified localized string id properties to load from same context as L()

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-13966

**Summary:**
- This is a follow-up PR to #10549.
- Modifies localized string ID properties such as `titleid` and `textid` to load from the same context as the `L()` function.
- This is needed since above mentioned PR modified `Ti.Locale.setLanguage()` to change the application context's locale/language instead of the root activity, which is needed so that app restarts will continue to use the selected language.
- Note that `L()` and image loading in the old code has always been loaded via the application context's resources. What has changed on Google's end is that the resource configuration reference is no longer shared between all contexts. This PR ensures all resources are loaded consistently.

**Test:**
_This test assumes device's default language is English._
1. Create a Classic Titanium app project.
2. Create an `i18n/en` folder and `i18n/es` folder under the project's root directory.
3. Copy the 2 "string.xml" below to the above folders.
4. Copy the below "app.js" code to its "Resources" folder.
5. Build and run on Android.
6. Verify app's top title bar reads `Hello World` in English.
7. Verify text under `'textid' label:` reads `Hello World` in English.
8. Verify text under `'L()' label:` reads `Hello World` in English.
9. Tap the "Select Spanish" button. (This calls `Ti.Locale.setLanguage("es")`.)
10. Tap the "Re-Open Window" button. (Will close window and re-open with new language.)
11. Verify app's top title bar reads `Hola Mundo` in Spanish.
12. Verify text under `'textid' label:` reads `Hola Mundo` in Spanish.
13. Verify text under `'L()' label:` reads `Hola Mundo` in Spanish.

---
**i18n/en/strings.xml**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<resources>
	<string name="hello_world">Hello World</string>
</resources>
```

---
**i18n/es/strings.xml**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<resources>
	<string name="hello_world">Hola Mundo</string>
</resources>
```

---
Resources/app.js
```javascript
function showMainWindow() {
	var window = Ti.UI.createWindow({
		titleid: "hello_world",
		layout: "vertical",
		extendSafeArea: false,
	});
	window.add(Ti.UI.createLabel({
		text: "'textid' label:",
		top: "20dp",
	}));
	window.add(Ti.UI.createLabel({
		textid: "hello_world",
		top: "5dp",
	}));
	window.add(Ti.UI.createLabel({
		text: "'L()' label:",
		top: "20dp",
	}));
	window.add(Ti.UI.createLabel({
		text: L("hello_world"),
		top: "5dp",
	}));
	var englishButton = Ti.UI.createButton({
		title: "Select English",
		top: "20dp",
	});
	englishButton.addEventListener("click", function() {
		Ti.Locale.setLanguage("en");
	});
	window.add(englishButton);
	var spanishButton = Ti.UI.createButton({
		title: "Select Spanish",
		top: "5dp",
	});
	spanishButton.addEventListener("click", function() {
		Ti.Locale.setLanguage("es");
	});
	window.add(spanishButton);
	var reopenButton = Ti.UI.createButton({
		title: "Re-Open Window",
		top: "20dp",
	});
	reopenButton.addEventListener("click", function() {
		window.exitOnClose = false;
		window.close();
		showMainWindow();
	});
	window.add(reopenButton);
	window.addEventListener("open", function() {
		Ti.API.info("@@@ Selected language: " + Ti.Locale.currentLanguage);
	});
	window.open();
}
showMainWindow();
```
